### PR TITLE
Fast conversion from ShortString to String

### PR DESCRIPTION
**Pull request by ScottPJones**

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -46,8 +46,22 @@ function ShortString{T}(s::ShortString{S}) where {T, S}
...
 
+const shft_int = UInt === UInt32 ? 2 : 3
 
-String(s::ShortString) = String(reinterpret(UInt8, [s.size_content|>ntoh])[1:sizeof(s)])
+function String(s::ShortString{T}) where {T}
+    len = sizeof(s)
+    len === 0 && return ""
+    val = ntoh(s.size_content)
+    sv = Base.StringVector(len)
+    # Loop over UInt64 sized chunks here
```
Comment (oxinabox) [inline src/base.jl:56]:
```suggestion
    # Loop over UInt sized chunks here
```

File: src/base.jl [hunk 2]
```diff
@@ -46,8 +46,22 @@ function ShortString{T}(s::ShortString{S}) where {T, S}
     ShortString{T}(content | T(sz))
 end
 
+const shft_int = UInt === UInt32 ? 2 : 3
```
Comment (oxinabox) [inline src/base.jl:49]: Docstring?

**Review by oxinabox — APPROVED** [review]
LGTM

File: src/base.jl (hunk 2 above)
Comment (xiaodaigh) [reply src/base.jl:49]: should constants be all capitalised? Like `SHFT_INT`

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "ShortStrings"
 uuid = "63221d1c-8677-4ff0-9126-0ff0817b4975"
 authors = ["Dai ZJ <zhuojia.dai@gmail.com>"]
```
Comment (xiaodaigh) [inline Project.toml:3]: Can update this too

File: src/base.jl (hunk 1 above)
Comment (xiaodaigh) [reply src/base.jl:56]: can we add a benchmark for both to prevent regression and point to that benchmark. So we can see how much faster this is?

**Pull request merged**